tools/prerelease-tool/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+type prereleaseTool struct {
+	Releases    []*github.RepositoryRelease
+	Limit       int
+	Owner       string
+	Repo        string
+	client      *github.Client
+	Prereleases []github.RepositoryRelease
+	DryRun      bool
+}
+
+func main() {
+	dryRun := flag.Bool("dry-run", false, "perform a dry run instead of executing create/delete/update actions")
+	limit := flag.Int("limit", 7, "limit of prereleases to keep")
+	owner := flag.String("owner", "filecoin-project", "github owner or organization")
+	repo := flag.String("repo", "go-filecoin", "github project repository")
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if !ok {
+		log.Fatal("Github token must be provided through GITHUB_TOKEN environment variable")
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+	flag.Parse()
+	r := prereleaseTool{
+		Limit:  *limit,
+		Owner:  *owner,
+		Repo:   *repo,
+		client: client,
+		DryRun: *dryRun,
+	}
+	if err := r.getReleases(ctx); err != nil {
+		log.Fatalf("Could not find any releases: %+v", err)
+	}
+	r.getPrereleases()
+	r.sortReleasesByDate()
```

Comment:
This pattern of an object having multiple state-mutating methods that must be called in order, and only one order that makes any sense, is a bit odd. It would be easier to use and understand with a single `DoIt()` method that can then call some functions/methods that take parameters and return values, rather than mutating shared state.

It's not a big deal if we don't expect the scope of this code to grow, I won't block on it.

Reply:
I did some method renaming and incorporated the sort method in to the getPreleases method. Right now the purpose of this is very tightly scoped but think it should be easy enough to adapt this if the scope grows.